docs: add purego info (#25)
tree-sitter/go-tree-sitter: README.md

go get github.com/tree-sitter/tree-sitter-javascript@latest
```

Alternatively you can also load grammars at runtime from a shared library via [purego](https://github.com/ebitengine/purego).

> This demonstrates how to load the JavaScript grammar from a shared library (`libtree-sitter-name.so`) at runtime on Linux & macOS:

Reviewer: ```suggestion
The example below shows how to load the JavaScript grammar from a shared library (`libtree-sitter-PARSER_NAME.so`) at runtime on Linux & macOS:
```

For more information on other platforms, see the [purego documentation](https://github.com/ebitengine/purego#supported-platforms)

```go
package main

import (
	tree_sitter "github.com/tree-sitter/go-tree-sitter"
	"github.com/ebitengine/purego"
)

func main() {

> 	lib, err := purego.Dlopen(path, purego.RTLD_NOW|purego.RTLD_GLOBAL)

Reviewer: Maybe add a dummy for path?
```suggestion
    path := "/path/to/your/parser.so"
	lib, err := purego.Dlopen(path, purego.RTLD_NOW|purego.RTLD_GLOBAL)
```

> 	var newLanguage func() uintptr
> 	purego.RegisterLibFunc(&newLanguage, "libtree-siiter-javascript.so", "tree_sitter_javascript")
> 
> 	language := tree_sitter.NewLanguage(unsafe.Pointer(newLanguage()))

Reviewer: Since I expect people to use many languages, let's call this javascriptLanguage()

```suggestion
	var javascriptLanguage func() uintptr
	purego.RegisterLibFunc(&javascriptLanguage, "libtree-siiter-javascript.so", "tree_sitter_javascript")

	language := tree_sitter.NewLanguage(unsafe.Pointer(javascriptLanguage()))
```

}
```

> Due to [bugs with `runtime.SetFinalizer` and CGO](https://groups.google.com/g/golang-nuts/c/LIWj6Gl--es), you must always call `Close`
> on an object that allocates memory from C. This must be done for the `Parser`, `Tree`, `TreeCursor`, `Query`, `QueryCursor`, and `LookaheadIterator` objects.

Reviewer: Can you do me a favor and emphasize this? Might be easy to miss as the readme grows. 

```suggestion
> [!NOTE]
> Due to [bugs with `runtime.SetFinalizer` and CGO](https://groups.google.com/g/golang-nuts/c/LIWj6Gl--es), you must always call `Close`
> on an object that allocates memory from C. This must be done for the `Parser`, `Tree`, `TreeCursor`, `Query`, `QueryCursor`, and `LookaheadIterator` objects.
```
